src/components/DpIcon/util/iconConfig.ts

```diff
@@ -0,0 +1,184 @@
+  'clock': PhClock,
+  'copy': PhCopy,
+  'file': PhFile,
+  'highlighter': PhHighlighter,
+  'hourglass': PhHourglass,
+  'info': PhInfo,
+  'plus': PhPlus,
+  'phone': PhPhone,
+  'question': PhQuestion,
+  'sign-out': PhSignOut,
+  'tag': PhTag,
+  'user': PhUser,
+  'users-three': PhUsersThree,
+  'warning': PhWarning,
+  'x': PhX,
+  'x-circle': PhXCircle
+}
+
+// icon names that are not used yet but should replace the aliases
+const mappedAliasedIconNames: Record<AliasedPhosphorIconName, Component> = {
+  'arrows-clockwise': PhArrowsClockwise,
+  'circle-notch': PhCircleNotch,
+  'clock-counter-clockwise': PhClockCounterClockwise,
+  'corners-in': PhCornersIn,
+  'corners-out': PhCornersOut,
+  'dots-six-vertical': PhDotsSixVertical,
+  'download-simple': PhDownloadSimple,
+  'envelope-simple': PhEnvelopeSimple,
+  'gear-six': PhGearSix,
+  'lock-simple': PhLockSimple,
+  'lock-simple-open': PhLockSimpleOpen,
+  'magnifying-glass': PhMagnifyingGlass,
+  'pencil-simple': PhPencilSimple,
+  'robot': PhRobot,
+  'warning-diamond': PhWarningDiamond,
+}
+
+// icon names that are used as aliases - the phosphor icon has a different name
+const mappedIconAliases: Record<IconAlias, Component> = {
+  'ai': PhRobot,
+  'cancel': PhX,
+  'chevron-down': PhArrowDown,
+  'chevron-left': PhArrowLeft,
+  'chevron-right': PhArrowRight,
+  'chevron-up': PhArrowUp,
+  'close': PhX,
+  'compress': PhCornersIn,
+  'delete': PhTrash,
+  'download': PhDownloadSimple,
+  'drag-handle': PhDotsSixVertical,
+  'edit': PhPencilSimple,
+  'expand': PhCornersOut,
+  'history': PhClockCounterClockwise,
+  'lock': PhLockSimple,
+  'mail': PhEnvelopeSimple,
+  'refresh': PhArrowsClockwise,
+  'remove': PhX,
+  'search': PhMagnifyingGlass,
+  'settings': PhGearSix,
+  'severe': PhWarningDiamond,
+  'spinner': PhCircleNotch,
+  'success': PhCheck,
+  'unlock': PhLockSimpleOpen,
+  'userSolid': PhUser,
+  'xmark': PhX
+}
+
+// ICON PROPORTIONS
+
+const iconsProportions: Record<string, IconProportion> = {
+  'arrow-down': 'portrait',
+  'arrow-left': 'landscape',
+  'arrow-right': 'landscape',
+  'arrow-up': 'portrait',
+  'caret-down': 'landscape',
+  'caret-left': 'portrait',
+  'caret-right': 'portrait',
+  'caret-up': 'landscape',
+  'caret-double-down': 'portrait',
+  'caret-double-up': 'portrait',
+  'check': 'landscape',
+  'dots-six-vertical': 'portrait',
+  'envelope-simple': 'landscape'
+}
+
+const aliasedIconsProportions: Record<string, IconProportion> = {
+  'chevron-down': 'landscape', // alias for caret-down
+  'chevron-left': 'portrait', // alias for caret-left
+  'chevron-right': 'portrait', // alias for caret-right
+  'chevron-up': 'landscape', // alias for caret-up
+  'drag-handle': 'portrait', // alias for dots-six-vertical
+  'mail': 'landscape', // alias for envelope-simple
+}
+
+// EXPORTS
+
+export const iconComponents: Record<IconName, Component> = {
+  ...mappedIconNames,
+  ...mappedAliasedIconNames,
+  ...mappedIconAliases
+}
```

Comment:
I'm afraid i don't understand the logic how iconComponents are calculated from the objects defined above. Maybe you could give me a walktrough here?

Reply:
In each of the objects, the icon name is mapped to the icon vue component. I created several objects to separate icon names that are already identical to the ones used by phosphor-icons/vue from icon names that are aliases (= not identical to the names used by phosphor-icons/vue) and icon names that are not currently used, but should replace the aliases in the future. Then these 3 objects are merged into one, so all icon names are available for use.
Not sure if that answers your question? If it does, maybe we could find better names for the variables?

Reply:
Thanks for the clarification, but i still do not understand why you differentiate between mappedIconNames and mappedAliasedIconNames. I'd just put everything in one object and have the aliases in the second.

Also, as all of these objects actually contain components already, i'd suggest to remove the "names" part:

```suggestion
export const iconComponents: Record<IconName, Component> = {
  ...mappedIcons, // Contains both mappedIconNames + mappedAliasedIconNames
  ...mappedAliasedIcons // Renamed from mappedIconAliases
}
```

Reply:
I separated them because I thought it would make things easier to understand, but maybe that's not really working :sweat_smile: Changed in 096c044e42096163c54465b37e559a52e20a4032